Barotrauma/BarotraumaShared/SharedSource/Items/ItemPrefab.cs

```diff
@@ -185,6 +186,21 @@ public FabricationRecipe(XElement element, ItemPrefab itemPrefab)
                         break;
                 }
             }
+
+            RecipeHash = GenerateHash();
+        }
+
+        private int GenerateHash()
+        {
+            var outputId = TargetItem.UIntIdentifier;
+
+            var requiredItems = string.Join(':', RequiredItems
+                .Select(i => i.ItemPrefabs.Select(p => p.UIntIdentifier))
+                .Select(i => string.Join(':', i)));
+
+            var requiredSkills = string.Join(':', RequiredSkills.Select(s => $"{s.Identifier}:{s.Level}"));
+
+            return $"{Amount}|{outputId}|{RequiredTime}|{requiredItems}|{requiredSkills}".GetHashCode();
```

Comment:
This should be `ToolBox.StringToUInt32Hash` instead of `string.GetHashCode`. This is because GetHashCode's implementation is not guaranteed to be stable; in our case, since we use .NET Core 3.1, that means that GetHashCode is not even stable across executions, which means that the chances of all clients generating the same hashes for the recipes here is next to none.